src/js/utils/api.ts

```diff
@@ -124,4 +126,20 @@ export const removeUrlParam = (key: string, search?: string) => {
   return params.toString();
 };
 
+export const getDeleteObjectUrl = (
+  objectType: ObjectTypes,
+  id: string,
+  userDeleteUrl: string | undefined,
+) => {
+  if (userDeleteUrl) {
+    return userDeleteUrl;
+  }
+  const urlFn = window.api_urls[`${objectType}_detail`];
+  let baseUrl;
+  if (urlFn && id) {
+    baseUrl = urlFn(id);
+  }
+  return baseUrl;
+};
```

Comment:
Personally I still think it would be worth getting rid of this and replacing it with a more generic `url` option in the `deleteObject` action, which could be set to `window.api_urls.current_user_detail()` in the one case where we want that.